### PR TITLE
Initial setup of project.

### LINKED ISSUE #1: Setup CI for running PoC unit tests
Probably use github actions/workflow. Initially just to run the unit tests, format, lint, security, coverage.

### PR DESCRIPTION
**Pull request by objectiser**
Fixes #1 with exception of codecov which needs to be installed, secret created, etc.

### CODE REVIEW

**Comment by objectiser** [comment]
@jpkrohling PR didn't trigger the github workflow - so assume project needs to be configured to enable it?

EDIT: Looks like it ran in my repo, but possibly doesn't run in this repo until merged - and then only for subsequent PRs?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+.PHONY: lint
+lint:
+	@echo Linting...
+	@./.ci/lint.sh
+
+.PHONY: security
+security:
+	@echo Security...
+	@${GOPATH}/bin/gosec -quiet -exclude=G104  ./... 2>/dev/null
```
Comment (jpkrohling) [inline Makefile]: For a new code base, perhaps run without exclusions?

File: .ci/lint.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+GOLINT=golint
+EMPTY="[[:space:]]"
```
Comment (jpkrohling) [inline .ci/lint.sh]: Just realized that this isn't being used.

**Review by jpkrohling — APPROVED** [review]
LGTM

**Comment by jpkrohling** [comment]
> EDIT: Looks like it ran in my repo, but possibly doesn't run in this repo until merged - and then only for subsequent PRs?

That's my guess as well

**Review by jpkrohling — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**